src/app/components/lineage-graph/lineage-graph.component.ts

```diff
@@ -23,10 +20,10 @@ import { LineageGraphNodeKind } from "src/app/dataset-view/additional-components
     styleUrls: ["./lineage-graph.component.scss"],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class LineageGraphComponent implements OnChanges, OnInit {
-    @Input() public links: Edge[];
-    @Input() public nodes: Node[];
-    @Input() public currentDataset: DatasetLineageBasicsFragment;
+export class LineageGraphComponent {
+    @Input() public links: MaybeNull<Edge[]>;
+    @Input() public nodes: MaybeNull<Node[]>;
+    @Input() public currentDataset: MaybeNull<DatasetLineageBasicsFragment>;
 
     @Output() public onClickNodeEvent = new EventEmitter<Node>();
```

Comment:
I think may of the flag variables below are readonly in nature.
Maybe we also need to group them into some "configuration" structure.

Reply:
Grouped into object

Reply:
Where? I still see

```

    public draggingEnabled = false;
    public panningEnabled = true;
    public zoomEnabled = true;
    public zoomSpeed = 0.03;
    public minZoomLevel = 0.3;
    public maxZoomLevel = 4.0;
    public panOnZoom = true;
    public autoZoom = true;
    public autoCenter = true;
    public showMiniMap = true;
```